### PR TITLE
cluster: detect bad member in self hosted case

### PR DESCRIPTION
**Pull request by xiang90**
this should help debugging https://github.com/coreos/etcd-operator/issues/978

### CODE REVIEW

**Comment by hongchaodeng** [comment]
@etcd-bot retest this please

**Comment by hongchaodeng** [comment]
@etcd-bot retest this please

known issue: https://github.com/coreos/etcd-operator/issues/978

**Comment by hongchaodeng** [comment]
Issue #978 happens quite often.
Let's wait https://github.com/coreos/etcd-operator/pull/999 to merge and then retest this.

**Comment by hongchaodeng** [comment]
@xiang90 
Can you rebase?

**Comment by xiang90** [comment]
@hongchaodeng rebased.

File: pkg/cluster/member.go [hunk 1]
```diff
@@ -38,6 +39,11 @@ func (c *Cluster) updateMembers(known etcdutil.MemberSet) error {
 				return errUnexpectedUnreadyMember
 			}
 
+			if !strings.HasPrefix(m.Name, c.cluster.Metadata.GetName()) {
```
Comment (hongchaodeng) [inline pkg/cluster/member.go]: This is not right.
The non-managed member could have a name not prefixed with cluster name.

Comment (xiang90) [inline pkg/cluster/member.go]: right. we need to compare the client url against the seed url. if it is one of the seed, we need to ignore it.

Comment (hongchaodeng) [inline pkg/cluster/member.go]: We actually just compare it below this.
Please put this check after that.

Comment (xiang90) [inline pkg/cluster/member.go]: fixed.

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**